dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/MetaDataController.kt

```diff
@@ -50,8 +58,44 @@ class MetaDataController(
         val currentUser = DatalandAuthentication.fromContextOrNull()
         val metaInfo = dataMetaInformationManager.getDataMetaInformationByDataId(dataId)
         if (!metaInfo.isDatasetViewableByUser(DatalandAuthentication.fromContextOrNull())) {
-            throw AccessDeniedException(logMessageBuilder.generateAccessDeniedExceptionMessage(metaInfo.qaStatus))
+            throw AccessDeniedException(
+                logMessageBuilder.generateAccessDeniedExceptionMessage(
+                    metaInfo.qaStatus,
+                ),
+            )
         }
         return ResponseEntity.ok(metaInfo.toApiModel(currentUser))
     }
+
+    override fun getInfoOnSourceabilityOfDataSets(
```

Comment:
This should include the information when (timestamp) and by whom (user-id) the data was marked as unsourcable. This information has to stored when the post endpoint is called.

Reply:
So you also want to filter for both?